File: client-scalaz/src/main/scala/org/openapitools/client/api/ContactInfo.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ContactInfo._

case class ContactInfo (
  homePhone: Option[String],
cellPhone: Option[String],
cellCarrier: Option[CellCarrier],
businessPhone: Option[String],
businessPhoneExt: Option[String],
faxNumber: Option[String],
timeZone: Option[String],
utcOffset: Option[String],
code501c3: Option[String],
emailAddress: Option[String],
address: Option[Address],
web: Option[String])

object ContactInfo {
  import DateTimeCodecs._

  implicit val ContactInfoCodecJson: CodecJson[ContactInfo] = CodecJson.derive[ContactInfo]
  implicit val ContactInfoDecoder: EntityDecoder[ContactInfo] = jsonOf[ContactInfo]
  implicit val ContactInfoEncoder: EntityEncoder[ContactInfo] = jsonEncoderOf[ContactInfo]
}

File: client-clojure/src/sirqul_io_t_platform/specs/contact_info.clj
(ns sirqul-io-t-platform.specs.contact-info
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.cell-carrier :refer :all]
            [sirqul-io-t-platform.specs.address :refer :all]
            )
  (:import (java.io File)))


(def contact-info-data
  {
   (ds/opt :homePhone) string?
   (ds/opt :cellPhone) string?
   (ds/opt :cellCarrier) cell-carrier-spec
   (ds/opt :businessPhone) string?
   (ds/opt :businessPhoneExt) string?
   (ds/opt :faxNumber) string?
   (ds/opt :timeZone) string?
   (ds/opt :utcOffset) string?
   (ds/opt :code501c3) string?
   (ds/opt :emailAddress) string?
   (ds/opt :address) address-spec
   (ds/opt :web) string?
   })

(def contact-info-spec
  (ds/spec
    {:name ::contact-info
     :spec contact-info-data}))
